### ISSUE TITLE
Transpose of a CKKSTensor

### ISSUE THREAD

**Issue by danif93** [OpenMined/TenSEAL#233]
## Feature Description
API extension that performs the matrix transposition of a CKKSTensor. Suppose that _m_ is a CKKSTensor with shape _(2,3)_, then _m.transpose()_ returns a new CKKSTensor _m_T_ (or alternatively modifies _m_ in-place) ) equal to the transpose of _m_ ( the shape of _m_T_ is _(3,2)_ with _m_T[i,j]=m[j,i]_).
Have a look to the following pseudo-code:
`m = CKKSTensor(context, [[1,2],[3,4],[5,6]])`
`assert m.shape==[3,2]`
`m_T = m.transpose() # or alternatively 'm.transpose_()' for inplace operation`
`assert (m_T.shape==[2,3]) and (m[i,j]==m_T[j,i] for any i in m.shape[0], j in m.shape[1])`

## Is your feature request related to a problem?
I am writing a neural network with privacy guarantees that leverages on homomorphic encryption during both the training and inference phases. I need the transpose of the CKKSTensor relative to the forward pass of a layer during gradients backpropagation.

## What alternatives have you considered?
I can extract the list of ciphertext from the CKKSTensor raw data. I can reorder that list for building a new CKKSTensor with the same context that is equal to the transpose of the original one. However, the creation of a CKKSTensor from raw data (list of ciphertext) is not publicly supported.

## Additional Context
- Maybe a `CKKSTensor.__init__(context, nested_list_of_ciphertexts)` should be supported for other similar applications;
- Look at [https://arxiv.org/abs/2012.13552](url) for an efficient implementation over MS Seal (the algorithm is outlined in the paper).
